### ISSUE TITLE
Feature: trusting actions from certain repositories

### ISSUE THREAD

**Issue by moe-ad** [woodruffw/zizmor#626]
### Pre-submission checks

- [x] I am **not** reporting a bug (crash, false positive/negative, etc). These must be filed via the bug report template.
- [x] I have looked through the open issues for a duplicate request.

### What's the problem this feature will solve?

Organizations do have internal actions repo they maintain (see [ansys/actions](https://github.com/ansys/actions)), and other repos within the same organization might want to "trust" these actions when running zizmor, especially when it comes to the `unpinned-uses` rule.

The current way of doing this is by adding an ignore comment in front of every call to an action from such a repo. Being able to specify trusted repos for the `unpinned-uses` rule once in `zizmor.yml` will be nice to have.

### Describe the solution you'd like

Something like the following would do:
```
zizmor.yml
rules:
  unpinned-uses:
    safe-repos:
      - ansys/actions
```

### Additional context

See the discussion in ansys/actions#725

**Comment by woodruffw** [comment]
Hey @moe-ad, thanks for opening an issue. This is somewhat similar to the policy idea being discussed in https://github.com/woodruffw/zizmor/issues/558, so please take a look at that as well!

**Comment by woodruffw** [comment]
Hey @moe-ad, sorry for the delay here! I forgot to respond earlier.

You're right that #558 is a bit different -- it defines an allow/denylist, rather than enabling different pinning policies. So it's not quite what you want.

However, the good news is that #558 inspired me to rethink `unpinned-uses` as well and, starting with v1.6.0, you'll be able to define this kind of policy! For example, here's how you might write a `zizmor.yml` to allow unpinned `uses` of `ansys/*`:

```yaml
rules:
  unpinned-uses:
    config:
      policies:
        ansys/*: any
```

See [the config docs](https://woodruffw.github.io/zizmor/audits/#unpinned-uses-configuration) for more notes on configuring this behavior.

**Comment by woodruffw** [comment]
Closing per #663.

**Comment by moe-ad** [comment]
> Hey [@moe-ad](https://github.com/moe-ad), sorry for the delay here! I forgot to respond earlier.
> 
> You're right that [#558](https://github.com/woodruffw/zizmor/issues/558) is a bit different -- it defines an allow/denylist, rather than enabling different pinning policies. So it's not quite what you want.
> 
> However, the good news is that [#558](https://github.com/woodruffw/zizmor/issues/558) inspired me to rethink `unpinned-uses` as well and, starting with v1.6.0, you'll be able to define this kind of policy! For example, here's how you might write a `zizmor.yml` to allow unpinned `uses` of `ansys/*`:
> 
> rules:
>   unpinned-uses:
>     config:
>       policies:
>         ansys/*: any
> See [the config docs](https://woodruffw.github.io/zizmor/audits/#unpinned-uses-configuration) for more notes on configuring this behavior.

Thanks for implementing this @woodruffw ! Great job!!!
